diff --git a/src/WebContent/bank/transaction.jsp b/src/WebContent/bank/transaction.jsp
index dc58e30..48f802b 100644
--- a/src/WebContent/bank/transaction.jsp
+++ b/src/WebContent/bank/transaction.jsp
@@ -32,6 +32,8 @@ IBM AltoroJ
     
 		<%@page import="java.util.Date"%>
 		<%@page import="com.ibm.security.appscan.altoromutual.model.Transaction"%>
+		<%@page import="com.ibm.security.appscan.altoromutual.util.ServletUtil" errorPage="notfound.jsp"%>
+
 		
 		<div class="fl" style="width: 99%;">
 		
@@ -123,9 +125,9 @@ IBM AltoroJ
 		<table border="0" style="padding-bottom:10px;">
 		    <tr>
 		        <td valign=top>After</td>
-		        <td><input id="startDate" name="startDate" type="text" value="<%=(request.getParameter("startDate")==null)?"":request.getParameter("startDate")%>"/><br /><span class="credit">yyyy-mm-dd</span></td>
+		        <td><input id="startDate" name="startDate" type="text" value="<%=(request.getParameter("startDate")==null)?"":ServletUtil.sanitizeWeb(request.getParameter("startDate"))%>"/><br /><span class="credit">yyyy-mm-dd</span></td>
 		        <td valign=top>Before</td>
-		        <td><input name="endDate" id="endDate" type="text" value="<%=(request.getParameter("endDate")==null)?"":request.getParameter("endDate") %>"/><br /><span class="credit">yyyy-mm-dd</span></td>
+		        <td><input name="endDate" id="endDate" type="text" value="<%=(request.getParameter("endDate")==null)?"":ServletUtil.sanitizeWeb(request.getParameter("endDate")) %>"/><br /><span class="credit">yyyy-mm-dd</span></td>
 		        <td valign=top><input type=submit value=Submit /></td>
 		    </tr>
 		</table>
